code/App.js

```diff
@@ -0,0 +1,174 @@
+import { FlatList, Pressable, Button } from "react-native";
+import * as WebBrowser from "expo-web-browser";
+import styled from "styled-components/native";
+import { API } from "./src/api/api";
+import TagPressable from "./src/components/TagPressable";
+
+const App = () => {
+  const [recipes, setRecipes] = useState([]);
+  const [selectedId, setSelectedId] = useState(0);
+  const [isLoading, setLoading] = useState(false);
+  const [showButton, setShowButton] = useState(false);
+
+  const flatListRef = useRef();
+
+  const scrollToTop = () => flatListRef?.current?.scrollToOffset({ animated: true, offset: 0 });
+
+  const handleOnPress = (url, index) => {
+    setLoading(true);
+    setSelectedId(index);
+    getRecipes(url);
+  };
+
+  const getRecipes = (url) => {
+    fetch(url)
+      .then((response) => response.json())
+      .then((data) => {
+        setRecipes(
+          data.response.results.map((item) => ({
+            id: item.id,
+            title: item.webTitle,
+            date: new window.Date(item.webPublicationDate).toLocaleString("en-UK", {
+              weekday: "short",
+              year: "numeric",
+              month: "short",
+              day: "numeric",
+            }),
+            thumbnail: item.fields.thumbnail,
+            url: item.webUrl,
+          }))
+        );
+        setLoading(false);
+        scrollToTop();
+      })
+      .catch((error) => console.log(error));
+  };
+
+  useEffect(() => getRecipes(API[0].url), []);
+
+  const renderItem = ({ item }) => {
+    return (
+      <Article key={item.id}>
+        <Thumbnail source={{ uri: item.thumbnail }} />
+        <Date>{item.date}</Date>
+        <Pressable onPress={() => WebBrowser.openBrowserAsync(item.url)}>
+          <Recipe>{item.title + " >"}</Recipe>
+        </Pressable>
+      </Article>
+    );
+  };
+
+  return (
+    <Container>
+      <Subtitle>RECIPES FROM</Subtitle>
+      <Title>The Guardian</Title>
+      <Tags>
+        {API.map((tag, index) => {
+          return (
+            <TagPressable
+              key={index}
+              title={tag.title}
+              url={tag.url}
+              index={index}
+              isSelected={index === selectedId}
+              onPress={handleOnPress}
+            />
+          );
+        })}
+      </Tags>
+      <SafeAreaViewStyled>
+        {isLoading && <Loader />}
+        <FlatList
+          data={recipes}
+          renderItem={renderItem}
+          key={(item) => item.id}
+          ref={flatListRef}
+          onScroll={(e) =>
+            e.nativeEvent.contentOffset.y > 600 ? setShowButton(true) : setShowButton(false)
+          }
+        />
+      </SafeAreaViewStyled>
+      {showButton && (
+        <Pressable onPress={() => scrollToTop()}>
+          <PressToTop>▲ Back to top</PressToTop>
+        </Pressable>
+      )}
+    </Container>
+  );
+};
+
+export default App;
+
+const Container = styled.View`
+  width: 325px;
+  margin: 0 auto;
+  padding: 50px 0;
+`;
```

Comment:
Perhaps minor thing but I believe I heard styling components should be placed above the exported function, we can doublecheck that